lib/lambda/delayedEmailProcessor.ts

```diff
@@ -0,0 +1,300 @@
+import { SQSEvent } from "aws-lambda";
+import { SESClient, SendEmailCommand, SendEmailCommandInput } from "@aws-sdk/client-ses";
+import { EmailAddresses, KafkaRecord, Events } from "shared-types";
+import { decodeBase64WithUtf8, getSecret } from "shared-utils";
+import { getEmailTemplates, getAllStateUsers } from "libs/email";
+import * as os from "libs/opensearch-lib";
+import { EMAIL_CONFIG, getCpocEmail, getSrtEmails } from "libs/email/content/email-components";
+import { htmlToText, HtmlToTextOptions } from "html-to-text";
+import pLimit from "p-limit";
+import { getOsNamespace } from "libs/utils";
+
+interface ProcessEmailConfig {
+  emailAddressLookupSecretName: string;
+  applicationEndpointUrl: string;
+  osDomain: string;
+  indexNamespace?: string;
+  region: string;
+  DLQ_URL: string;
+  userPoolId: string;
+  configurationSetName: string;
+  isDev: boolean;
+}
+
+/**
+ * Main SQS handler: parse each SQS message to get the original Kafka payload,
+ * then do OpenSearch lookups and send emails.
+ */
+export const handler = async (event: SQSEvent) => {
+  const requiredEnvVars = [
+    "emailAddressLookupSecretName",
+    "applicationEndpointUrl",
+    "osDomain",
+    "region",
+    "DLQ_URL",
+    "userPoolId",
+    "isDev",
+    "configurationSetName",
+  ] as const;
+
+  const missing = requiredEnvVars.filter((e) => !process.env[e]);
+  if (missing.length > 0) {
+    throw new Error(`Missing environment variables: ${missing.join(", ")}`);
+  }
+
+  const emailConfig: ProcessEmailConfig = {
+    emailAddressLookupSecretName: process.env.emailAddressLookupSecretName!,
+    applicationEndpointUrl: process.env.applicationEndpointUrl!,
+    osDomain: `https://${process.env.osDomain!}`,
+    indexNamespace: process.env.indexNamespace,
+    region: process.env.region!,
+    DLQ_URL: process.env.DLQ_URL!,
+    userPoolId: process.env.userPoolId!,
+    configurationSetName: process.env.configurationSetName!,
+    isDev: process.env.isDev === "true",
+  };
+
+  for (const sqsRecord of event.Records) {
+    try {
+      const kafkaRecord = JSON.parse(sqsRecord.body) as KafkaRecord;
+
+      const { key, value } = kafkaRecord; // sanity check
+      if (!key || !value) {
+        console.log("No key/value present. Possibly a tombstone or invalid data.");
+        continue;
+      }
```

Comment:
You can make a guard function that narrows the `sqsRecord.body` to a `KafkaRecord`

```ts
const isKafkaRecord = (parsedBody: unknown): parsedBody is KafkaRecord => /* logic pertaining to kafkaRecord, i.e. typeof parsedBody === 'object' && parsedBody?.value */ 
```

Reply:
But that's a nit